pilot/pkg/model/context.go

```diff
@@ -119,6 +120,54 @@ type Proxy struct {
 
 	// labels associated with the workload
 	WorkloadLabels config.LabelsCollection
+
+	// Istio version associated with the Proxy
+	IstioVersion *IstioVersion
+}
+
+var (
+	istioVersionRegexp = regexp.MustCompile(`^([1-9]+)\.([0-9]+)(\.([0-9]+))?`)
+)
+
+// IstioVersion encodes the Istio version of the proxy. This is a low key way to
+// do semver style comparisons and generate the appropriate envoy config
```

Comment:
why a "low key" way? Can we just use the semver library to do this for us?

Reply:
Because that library insists on parsing regexes from the string.. we dont want to run those regexes again and again. Secondly, we also need a way for consumers to do GE12, GE13 etc. which would involve more string parsing when used internally (like listener saying >=1.1) and the semver doign regexes, and finally matching..